x-pack/plugins/security_solution/common/endpoint/data_generators/endpoint_metadata_generator.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DeepPartial } from 'utility-types';
+import { merge } from 'lodash';
+import { gte } from 'semver';
+import { BaseDataGenerator } from './base_data_generator';
+import type { HostMetadataInterface, OSFields } from '../types';
+import { EndpointStatus, HostPolicyResponseActionStatus } from '../types';
+
+/**
+ * Metadata generator for docs that are sent by the Endpoint running on hosts
+ */
+export class EndpointMetadataGenerator extends BaseDataGenerator {
+  /** Generate an Endpoint host metadata document */
+  generate(overrides: DeepPartial<HostMetadataInterface> = {}): HostMetadataInterface {
+    const ts = overrides['@timestamp'] ?? new Date().getTime();
+    const hostName = this.randomHostname();
+    const agentVersion = overrides?.agent?.version ?? this.randomVersion();
+    const agentId = this.seededUUIDv4();
+    const isIsolated = this.randomBoolean(0.3);
+    const capabilities = ['isolation'];
+
+    // v8.4 introduced additional endpoint capabilities
+    if (gte(agentVersion, '8.4.0')) {
+      capabilities.push('kill_process', 'suspend_process', 'running_processes');
```

Comment:
I think there's a `RESPONDER_CAPABILITIES` const that you can use here instead.

Reply:
Ahh. yes. But: I'm wondering if I should use it. As we add more capabilities to it, we don't want them to automatically be added if agent version if > 8.4. Any additional one that we add should be included in the mock if the version is at least the version when it was introduced.

Reply:
Okay, I see what you mean. In that case, this is alright.